### PR TITLE
controlpickermenu: add fx parameter buttons & other missing controls

### PR DESCRIPTION
**Pull request by ronso0**
Add missing controls to controlpicker menu.
Continuation of #3093
https://bugs.launchpad.net/mixxx/+bug/1895364

- [x] add fx parameter buttons
- [x] fix Fx mix mode buttons
- [x] add some common skin controls
- [x] add Broadcast toggle
- [x] add Recording toggle

### CODE REVIEW

**Comment by ronso0** [comment]
I extend the mapping error message:
image

**Comment by ronso0** [comment]
I consider this ready to be merged, please check!

**Comment by Holzhaus** [comment]
There are a bunch of pre-commit issues. Please resolve them.

**Comment by ronso0** [comment]
> There are a bunch of pre-commit issues. Please resolve them.

Done. Those were changes that make the file inconsistent style-wise, but hej...
Ready!

**Comment by Holzhaus** [comment]
There are merge conflicts now.

**Comment by Holzhaus** [comment]
I'm still on vacation, if you want a quick review better request it from someone else ;-)

**Comment by ronso0** [comment]
thanks for letting me know!

**Review by Holzhaus — APPROVED** [review]
LGTM!

File: src/controllers/dlgcontrollerlearning.cpp [hunk 1]
```diff
@@ -466,13 +466,18 @@ void DlgControllerLearning::controlClicked(ControlObject* pControl) {
...
-                       "This could be because you are using an old skin"
-                       " and this control is no longer supported.\n"
-                       "\nYou tried to learn: %1,%2").arg(key.group, key.item),
-                    QMessageBox::Ok, QMessageBox::Ok);
+                this,
+                Version::applicationName(),
+                tr("The control you clicked in Mixxx is not learnable.\n"
+                   "This could be because you are either using an old skin"
+                   " and this control is no longer supported.\n"
+                   "Or you clicked a control that provides visual feedback"
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/controllers/dlgcontrollerlearning.cpp]: Done.

**Comment by Holzhaus** [comment]
There's a merge conflict

File: src/controllers/controlpickermenu.cpp [hunk 2]
```diff
@@ -767,20 +826,55 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
...
                tr("Effect Rack Show/Hide"),
                tr("Show/hide the effect rack"), guiMenu);
-    addControl("[Library]", "show_coverart",
-               tr("Cover Art Show/Hide"),
-               tr("Show/hide cover art"), guiMenu);
+    addControl("[Skin]",
+            "show_4effectunits",
+            tr("4 Effect Units Show/Hide"),
+            tr("Switches between showing 2 and 4 effect units"),
+            guiMenu);
```
Comment (Holzhaus) [inline src/controllers/controlpickermenu.cpp:833]: Do these exist for all skins?

Comment (ronso0) [reply src/controllers/controlpickermenu.cpp:833]: all except Shade ;)
We have one case IINM where it's the other way around: show_spinny1 or something that only exists in Shade, in my recent MIDI PR.
I added a comment to place a hint about this in the Wizard dialog.

File: src/controllers/controlpickermenu.cpp [hunk 3]
```diff
@@ -631,8 +651,9 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
                            tr("Switch to either next or previous effect"),
                            effectSlotMenu, false, slotDescriptionPrefix);
 
-                const int numParameterSlots = static_cast<int>(ControlObject::get(
-                        ConfigKey(effectSlotGroup, "num_parameterslots")));
+                // Effect parameter knobs
+                const int numParameterSlots = ControlObject::get(
```
Comment (Holzhaus) [inline src/controllers/controlpickermenu.cpp:655]:
```
src/controllers/controlpickermenu.cpp:655:65: warning: conversion from ‘double’ to ‘int’ may change value [-Wfloat-conversion]
  655 |                 const int numParameterSlots = ControlObject::get(
      |                                               ~~~~~~~~~~~~~~~~~~^
  656 |                         ConfigKey(effectSlotGroup, "num_parameterslots"));
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: src/controllers/controlpickermenu.cpp [hunk 4]
```diff
@@ -669,6 +690,37 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
                                parameterDescriptionPrefix);
 
                 }
+
+                // Effect parameter buttons
+                const int numButtonParameterSlots = ControlObject::get(
```
Comment (Holzhaus) [inline src/controllers/controlpickermenu.cpp:695]:
```
src/controllers/controlpickermenu.cpp:695:71: warning: conversion from ‘double’ to ‘int’ may change value [-Wfloat-conversion]
  695 |                 const int numButtonParameterSlots = ControlObject::get(
      |                                                     ~~~~~~~~~~~~~~~~~~^
  696 |                         ConfigKey(effectSlotGroup, "num_button_parameterslots"));
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: src/controllers/controlpickermenu.cpp (hunk 3 above)
Comment (ronso0) [reply src/controllers/controlpickermenu.cpp:655]: so this was the merge conflict..didn't look close enough.
reverted.

File: src/controllers/controlpickermenu.cpp (hunk 4 above)
Comment (ronso0) [reply src/controllers/controlpickermenu.cpp:695]: reverted as well.

**Comment by Holzhaus** [comment]
There are merge conflicts now.

**Comment by ronso0** [comment]
Ready again!

**Comment by ronso0** [comment]
Draft again until some can clairfy my `black` fixes.
from https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/pre-commit.20.2F.20yaml.20.2F.20black
> I repeatedly noticed CI pre-commit complaints about some python files (tools/clang_format.py, tools/serato_heartbeat.py).
I did `pre-commit autoupdate` and then the issues were discovered locally, as well.
But that changed the yaml file, too, changing some hook versions from `stable` to `v.1.7.x` for example.
How to deal with that? What's the recommended way to keep pre-commit up to date?
>
>https://github.com/pre-commit/pre-commit/issues/1354
doesn't help me much...

**Comment by ronso0** [comment]
Ready again!

**Comment by ronso0** [comment]
> Ready again!

_ping_

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you!

**Pull request merged**